.changeset/ten-cougars-pretend.md

```diff
@@ -0,0 +1,69 @@
+---
+'@astrojs/netlify': major
+---
+
+# Netlify Adapter v4 simplifies static + SSR deployments
+
+This update is a complete overhaul of the Netlify adapter.
+It simplifies the user-facing config, and resolves a number of bugs along the way.
+
+Here's what changes:
+
+## Netlify Context is automatically available via Locals
+
+In v3, you could use `netlify-edge-middleware.ts` to inject data from the Netlify context into your Astro locals.
+In v4, this file is no longer needed because the Netlify context is automatically made available via `Astro.locals.netlify.context`.
+You can use this context to access information about the user (like geolocation or IP address), your Netlify site (like deploy ID) or the request (like its request ID or the CDN region it's served from).
+
+**Action Required:**
+Remove the `netlify-edge-middleware.ts` or `netlify-edge-middleware.js` file.
+In your codebase, change all usage of locals injected through that file to use `Astro.locals.netlify.context` instead.
+
+### Image CDN
+
+v4 of this adapter integrates your Astro site with Netlify [Image CDN](https://docs.netlify.com/image-cdn/overview/).
+This allows transforming images on-the-fly without impacting build times.
+It's implemented using an [Astro Image Service](https://docs.astro.build/en/reference/image-service-reference/), and enabled by default.
+
+## On-Demand Builders are no longer supported
+
+On-Demand Builders (ODB) allows SSR-Rendered pages to be cached using a Time to Live (TTL) strategy.
+While the Netlify platform continues to support existing pages with ODBs, there's now the much more powerful
+[Fine-Grained Cache Control](https://www.netlify.com/blog/swr-and-fine-grained-cache-control) that's recommended going forward.
```

Comment:
```suggestion
## Replacement for On-Demand Builders

On-Demand Builders (ODB) allows SSR-Rendered pages to be cached using a Time to Live (TTL) strategy.
While the Netlify platform continues to support existing pages with ODBs, we now recommend using the much more powerful
[Fine-Grained Cache Control](https://www.netlify.com/blog/swr-and-fine-grained-cache-control) going forward.
```
I still think the text contradicted the header, so I'd suggest something closer to this. But this is much better! If you have a specific link for updating/converting, I'd also suggest adding a line like:

> Read [Netlify's documentation for converting your on-demand builders]() to upgrade to new the recommended caching method.

Reply:
That's good, thank you! We don't have specific docs for that, sadly, so there's nothing to link to :/